doc/model.rst

```diff
@@ -0,0 +1,683 @@
+-  `devices_per_node`: 2 - Use two GPUs per node.
+-  `first_device`: 0 - Start with 0 index GPU.
+
+When the model is started via ``Experiment.start()``, the TF function will be loaded to the
+standard orchestrator that is launched prior to the start of the model.
+
+-------------------
+TorchScript Scripts
+-------------------
+SmartSim supports the execution of TorchScripts
+with Models. Regardless of whether
+the orchestrator is colocated or not, each script added to the Model is
+loaded into the orchestrator prior to the execution of entity.
+Users have the flexibility to choose
+between `"GPU"` or `"CPU"` for device selection, and in environments with multiple
+devices, specific device numbers can be specified via `devices_per_node` such as `"GPU:1,"`.
+
+When specifying a TorchScript using ``Model.add_script()``, the
+following arguments are offered:
+
+- name (str): key to store script under
+- script (str, optional): TorchScript code (only supported for non-colocated orchestrators)
+- script_path (str, optional): path to TorchScript code
+- device (str, optional): device for script execution, defaults to “CPU”
+- devices_per_node (int): The number of GPU devices available on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+- first_device (int): The first GPU device to use on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+
+You might use TorchScript scripts to represent individual models within the model.
+
+Example: Loading an TorchScript to the Model
+--------------------------------------------
+In this example, we demonstrate how to instruct SmartSim to load
+an TorchScript into the database at model runtime.
+
+**Python Script: Define a TorchScript for Model Purposes**
+
+Define the TorchScript code to a variable in the Python driver script:
+
+.. code-block:: python
+
+    torch_script_str = "def negate(x):\n\treturn torch.neg(x)\n"
+
+**SmartSim Model Integration:**
+
+Assuming an initialized ``Model`` named `smartsim_model`, we specify the
+following parameters to the ``Model.add_script()`` function:
+
+.. code-block:: python
+
+    smartsim_model.add_script(name="example_script", script=torch_script_str, device="GPU", devices_per_node=2, first_device=0)
+
+In this integration, we provide the following details:
+
+-  `name`: "example_script" - key to store script under
+-  `script`: torch_script_str - TorchScript code
+-  `device`: "CPU" - device for script execution
+-  `devices_per_node`: 2 - Use two GPUs per node.
+-  `first_device`: 0 - Start with 0 index GPU.
+
+When the model is started via ``Experiment.start()``, the TorchScript will be loaded to the
+orchestrator that is launched prior to the start of the model.
+
+=========================
+Data Collision Prevention
+=========================
+--------
+Overview
+--------
+SmartSim and SmartRedis can avoid key collision by prepending program-unique
```

Comment:
what is "program-unique"? consider using alternate terminology (model-unique?) since this is the first time "program" has entered vocabulary.